framework/core/js/src/common/components/SplitDropdown.tsx

```diff
@@ -0,0 +1,56 @@
+import Dropdown, { IDropdownAttrs } from './Dropdown';
+import Button from './Button';
+import icon from '../helpers/icon';
+import Mithril from 'mithril';
+import classList from '../utils/classList';
+
+export interface ISplitDropdownAttrs extends IDropdownAttrs {}
+
+/**
+ * The `SplitDropdown` component is similar to `Dropdown`, but the first child
+ * is displayed as its own button prior to the toggle button.
+ */
+export default class SplitDropdown extends Dropdown {
+  static initAttrs(attrs: ISplitDropdownAttrs) {
+    super.initAttrs(attrs);
+
+    attrs.className = classList(attrs.className, 'Dropdown--split');
+    attrs.menuClassName = classList(attrs.menuClassName, 'Dropdown-menu--right');
+  }
+
+  getButton(children: Mithril.ChildArray): Mithril.Vnode<any, any> {
+    // Make a copy of the attrs of the first child component. We will assign
+    // these attrs to a new button, so that it has exactly the same behaviour as
+    // the first child.
+    const firstChild = this.getFirstChild(children);
+    const buttonAttrs = Object.assign({}, firstChild?.attrs);
```

Comment:
```suggestion
    const buttonAttrs = Object.assign({}, firstChild?.attrs ?? {});
```

Is this necessary?

Reply:
Why not

```
{ ...firstChild?.attrs ?? {} }
```

Reply:
I don't think this is necessary, the `getFirstChild` method is typed to explicitly return a `Vnode`. if somehow it returns a non `Vnode` then it should probably be fixed at that level?